src/modules/commander/gyro_calibration.cpp

```diff
@@ -304,29 +304,27 @@ int do_gyro_calibration(orb_advert_t *mavlink_log_pub)
 			}
 
 			calibration.set_calibration_index(uorb_index);
-			calibration.ParametersSave();
-		}
-
-		if (failed) {
-			calibration_log_critical(mavlink_log_pub, "ERROR: failed to set offset params");
-			res = PX4_ERROR;
-		}
-	}
 
-	param_notify_changes();
+			if (calibration.ParametersSave()) {
+				success = true;
```

Comment:
This won't be correct if the first iteration succeeds, but the next one fails.

Reply:
Thanks, fixed.